File: 7. Reverse Integer/7.py
class Solution:
    def reverse(self, x):
            """
            :type x: int
            :rtype: int
            """
            sign = lambda x: x and (1, -1)[x < 0]
            r = int(str(sign(x)*x)[::-1])
            return (sign(x)*r, 0)[r > 2**31 - 1]

File: 7. Reverse Integer/7.cpp
class Solution {
public:
    int reverse(int x) {
        long long j = 0;
    	do{
    		j = j * 10 + x % 10;
    	}while (x /= 10);
        return (j > INT_MAX || j < INT_MIN) ? 0: (int)j;
    }
};
